keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,166 @@ Running this set of tests is optional for a per-provider basis. We would
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CSIMigration, CSIMigration{cloud-provider}
+    - Components depending on the feature gate: kubelet, kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  Yes and No. If only CSIMigration feature flag is enabled, nothing will change on the cluster behavior. However, if CSIMigration && CSIMigration{cloud-provider} are both enabled, the behavior will change. The in-tree volume plugin that the cloud-provider use will be redirect to use the corresponding CSI driver. But from a user perspective, nothing will be noticed.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. Once there is an error or something unexpected, users can disable the CSI migration feature flags to roll it back.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+The CSI migration will continue to function.
+
+* **Are there any tests for feature enablement/disablement?**
+We have CSI Migration e2e test for each plugin that are implemented and maintained by each driver maintainer. For core K8s, we have unit tests including but not limited to:
+   - `pkg/volume/csimigration/plugin_manager_test.go`
+   - All unit tests in the csi-translation-lib `staging/src/k8s.io/csi-translation-lib/translate_test.go`
+   - Controller test with Migration on CSI sidecars: external-provisioner, external-resizer
+     - provisioner: pkg/controller/controller_test.go#TestProvisionWithMigration
+     - resizer: pkg/resizer/csi_resizer_test.go#TestResizeMigratedPV
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**  - The rollout can fail if the ordering of CSIMigration{cloud-provider} flag was wrongly enabled on kubelet and kube-controller-manager. Specifically, if on the node side kubelet enables the flag and control-plane side the flag is not enabled, then the volume will not be able to be mounted successfully. 
+    - For workloads that running on nodes have not enable CSI migration, those pods will not be impacted. 
+    - For any pod that is being rescheduled to nodes that turned on migration, the volume mount will fail and pod will not come up correctly.
```

Comment:
There is no pod rescheduling in Kubernetes - once bound to the node the pods lives there until EOL.

Can you clarify what you meant here?


[nit - it would be much easier to comment if the lines would be splitted]

Reply:
There is a restriction on enabling the feature gate. It requires node that being upgrade be drained before restarting kubelet. So when the node is being drained, there may or may not be pod movement and that could fail potentially with the wrong ordering of feature gate enablement.

Reply:
What I'm saying is "there is NOT  anything like 'pod movement' in Kubernetes".
Once pod is bounded to the node - it will never be moved to a different node.

What we do is delte a pod and create a new one. But it's not pod movement or rescheduling.

Reply:
Done